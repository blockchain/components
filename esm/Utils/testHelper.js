import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}