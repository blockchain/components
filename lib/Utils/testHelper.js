"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}