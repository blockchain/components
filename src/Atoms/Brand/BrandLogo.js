// @flow strict
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

type PropsType = {
  +size?: string,
}

const Wrapper = styled.svg.attrs({
  viewBox: '0 0 103.9 103.9',
})`
  height: ${(props) => props.size};
  width: ${(props) => props.size};
`
const Path1 = styled.path`
  fill: #00aee6;
`
const Path2 = styled.path`
  fill: #123962;
`
const Path3 = styled.path`
  fill: #b1d4e5;
`
const Polygon1 = styled.polygon`
  fill: #b1d4e5;
`
const Polygon2 = styled.polygon`
  fill: #2754ba;
`
const Path4 = styled.path`
  fill: #799eb2;
`

const BrandLogo = (props: PropsType) => (
  <Wrapper {...props}>
    <Path1 d="M38.6,52L17,73.5l27.1,27.1c4.4,4.4,11.4,4.4,15.8,0l27.1-27.1L65.3,52L52,38.6L38.6,52z" />
    <Path2 d="M17,30.4L3.3,44.1c-4.4,4.4-4.4,11.4,0,15.8L17,73.5L38.6,52L52,38.6L30.4,17L17,30.4z" />
    <Path3 d="M73.5,60.1l13.4,13.4l13.7-13.7c4.4-4.4,4.4-11.4,0-15.8l-5.5-5.5L73.5,60.1z" />
    <Polygon1 points="65.3,52 65.3,52 52,38.6 73.5,17 73.5,17 52,38.6 	" />
    <Polygon2 points="52,38.6 65.3,52 73.5,60.1 95.1,38.6 86.9,30.4 73.5,17 	" />
    <Path4 d="M73.5,17L59.8,3.3c-4.4-4.4-11.4-4.4-15.8,0L30.4,17L52,38.6L73.5,17z" />
  </Wrapper>
)

BrandLogo.propTypes = {
  size: PropTypes.string,
}

BrandLogo.defaultProps = {
  size: '100px',
}

export default BrandLogo
