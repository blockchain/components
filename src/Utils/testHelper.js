import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
