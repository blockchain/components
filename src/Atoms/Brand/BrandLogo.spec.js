import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import BrandLogo from './BrandLogo'

describe('BrandLogo', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<BrandLogo />)

    expect(tree.find(BrandLogo)).toMatchSnapshot()
  })

  it('should render correctly with size', () => {
    const tree = mountWithTheme(<BrandLogo size="100px" />)

    expect(tree.find(BrandLogo)).toMatchSnapshot()
  })
})
