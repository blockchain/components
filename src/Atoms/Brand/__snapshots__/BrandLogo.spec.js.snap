// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo should render correctly 1`] = `
.c0 {
  height: 100px;
  width: 100px;
}

.c1 {
  fill: #00aee6;
}

.c2 {
  fill: #123962;
}

.c3 {
  fill: #b1d4e5;
}

.c4 {
  fill: #b1d4e5;
}

.c5 {
  fill: #2754ba;
}

.c6 {
  fill: #799eb2;
}

<BrandLogo
  size="100px"
>
  <styled.svg
    size="100px"
  >
    <svg
      className="c0"
      size="100px"
      viewBox="0 0 103.9 103.9"
    >
      <styled.path
        d="M38.6,52L17,73.5l27.1,27.1c4.4,4.4,11.4,4.4,15.8,0l27.1-27.1L65.3,52L52,38.6L38.6,52z"
      >
        <path
          className="c1"
          d="M38.6,52L17,73.5l27.1,27.1c4.4,4.4,11.4,4.4,15.8,0l27.1-27.1L65.3,52L52,38.6L38.6,52z"
        />
      </styled.path>
      <styled.path
        d="M17,30.4L3.3,44.1c-4.4,4.4-4.4,11.4,0,15.8L17,73.5L38.6,52L52,38.6L30.4,17L17,30.4z"
      >
        <path
          className="c2"
          d="M17,30.4L3.3,44.1c-4.4,4.4-4.4,11.4,0,15.8L17,73.5L38.6,52L52,38.6L30.4,17L17,30.4z"
        />
      </styled.path>
      <styled.path
        d="M73.5,60.1l13.4,13.4l13.7-13.7c4.4-4.4,4.4-11.4,0-15.8l-5.5-5.5L73.5,60.1z"
      >
        <path
          className="c3"
          d="M73.5,60.1l13.4,13.4l13.7-13.7c4.4-4.4,4.4-11.4,0-15.8l-5.5-5.5L73.5,60.1z"
        />
      </styled.path>
      <styled.polygon
        points="65.3,52 65.3,52 52,38.6 73.5,17 73.5,17 52,38.6 	"
      >
        <polygon
          className="c4"
          points="65.3,52 65.3,52 52,38.6 73.5,17 73.5,17 52,38.6 	"
        />
      </styled.polygon>
      <styled.polygon
        points="52,38.6 65.3,52 73.5,60.1 95.1,38.6 86.9,30.4 73.5,17 	"
      >
        <polygon
          className="c5"
          points="52,38.6 65.3,52 73.5,60.1 95.1,38.6 86.9,30.4 73.5,17 	"
        />
      </styled.polygon>
      <styled.path
        d="M73.5,17L59.8,3.3c-4.4-4.4-11.4-4.4-15.8,0L30.4,17L52,38.6L73.5,17z"
      >
        <path
          className="c6"
          d="M73.5,17L59.8,3.3c-4.4-4.4-11.4-4.4-15.8,0L30.4,17L52,38.6L73.5,17z"
        />
      </styled.path>
    </svg>
  </styled.svg>
</BrandLogo>
`;

exports[`BrandLogo should render correctly with size 1`] = `
.c0 {
  height: 100px;
  width: 100px;
}

.c1 {
  fill: #00aee6;
}

.c2 {
  fill: #123962;
}

.c3 {
  fill: #b1d4e5;
}

.c4 {
  fill: #b1d4e5;
}

.c5 {
  fill: #2754ba;
}

.c6 {
  fill: #799eb2;
}

<BrandLogo
  size="100px"
>
  <styled.svg
    size="100px"
  >
    <svg
      className="c0"
      size="100px"
      viewBox="0 0 103.9 103.9"
    >
      <styled.path
        d="M38.6,52L17,73.5l27.1,27.1c4.4,4.4,11.4,4.4,15.8,0l27.1-27.1L65.3,52L52,38.6L38.6,52z"
      >
        <path
          className="c1"
          d="M38.6,52L17,73.5l27.1,27.1c4.4,4.4,11.4,4.4,15.8,0l27.1-27.1L65.3,52L52,38.6L38.6,52z"
        />
      </styled.path>
      <styled.path
        d="M17,30.4L3.3,44.1c-4.4,4.4-4.4,11.4,0,15.8L17,73.5L38.6,52L52,38.6L30.4,17L17,30.4z"
      >
        <path
          className="c2"
          d="M17,30.4L3.3,44.1c-4.4,4.4-4.4,11.4,0,15.8L17,73.5L38.6,52L52,38.6L30.4,17L17,30.4z"
        />
      </styled.path>
      <styled.path
        d="M73.5,60.1l13.4,13.4l13.7-13.7c4.4-4.4,4.4-11.4,0-15.8l-5.5-5.5L73.5,60.1z"
      >
        <path
          className="c3"
          d="M73.5,60.1l13.4,13.4l13.7-13.7c4.4-4.4,4.4-11.4,0-15.8l-5.5-5.5L73.5,60.1z"
        />
      </styled.path>
      <styled.polygon
        points="65.3,52 65.3,52 52,38.6 73.5,17 73.5,17 52,38.6 	"
      >
        <polygon
          className="c4"
          points="65.3,52 65.3,52 52,38.6 73.5,17 73.5,17 52,38.6 	"
        />
      </styled.polygon>
      <styled.polygon
        points="52,38.6 65.3,52 73.5,60.1 95.1,38.6 86.9,30.4 73.5,17 	"
      >
        <polygon
          className="c5"
          points="52,38.6 65.3,52 73.5,60.1 95.1,38.6 86.9,30.4 73.5,17 	"
        />
      </styled.polygon>
      <styled.path
        d="M73.5,17L59.8,3.3c-4.4-4.4-11.4-4.4-15.8,0L30.4,17L52,38.6L73.5,17z"
      >
        <path
          className="c6"
          d="M73.5,17L59.8,3.3c-4.4-4.4-11.4-4.4-15.8,0L30.4,17L52,38.6L73.5,17z"
        />
      </styled.path>
    </svg>
  </styled.svg>
</BrandLogo>
`;
